### ISSUE TITLE
Stub request does not work when request header is not string 

### ISSUE THREAD

**Issue by felipecdo** [bblimke/webmock#724]
When I am using Stub request as following, webmock says that it is a "Unregistered request". But check that it is the same value `{"My-Header" => 99}` and `req['My-Header'] = 99`

```
stub_request(:post, "www.example.com").
with(body: /world$/, headers: {"My-Header" => 99}).
to_return(body: "abc")

uri = URI.parse('http://www.example.com/')
req = Net::HTTP::Post.new(uri.path)
req['My-Header'] = 99

res = Net::HTTP.start(uri.host, uri.port) do |http|
  http.request(req, 'hello world')
end
```
        
Running in rspec, it throws:
```
     Failure/Error: http.request(req, 'hello world')
     WebMock::NetConnectNotAllowedError:
       Real HTTP connections are disabled. Unregistered request: POST http://www.example.com/ with body 'hello world' with headers {'Accept'=>'*/*', 'Accept-Encoding'=>'gzip;q=1.0,deflate;q=0.6,identity;q=0.3', 'My-Header'=>'99', 'User-Agent'=>'Ruby'}
       
       You can stub this request with the following snippet:
       
       stub_request(:post, "http://www.example.com/").
         with(body: "hello world",
              headers: {'Accept'=>'*/*', 'Accept-Encoding'=>'gzip;q=1.0,deflate;q=0.6,identity;q=0.3', 'My-Header'=>'99', 'User-Agent'=>'Ruby'}).
         to_return(status: 200, body: "", headers: {})
       
       registered request stubs:
       
       stub_request(:post, "http://www.example.com/").
         with(body: /world$/,
              headers: {'My-Header'=>'99'})
```

If I try to use `with(body: /world$/, headers: {"My-Header" => '99'}).` (as string) and it also raises exception.

Version: 'webmock', '3.0.1'

**Comment by bblimke** [comment]
Thanks for reporting.

Fixed and released as version 3.1.0

**Comment by felipecdo** [comment]
Thank you so much! Tested and it is ok!
